### PR TITLE
Improve management of credentials

### PR DESCRIPTION
**Pull request by podcastinator**
This PR:
1. Adds helper scripts to generate: (i) keys + certs for clients; and (ii) keys + cert for the signing authority (this authority signs the enclave binary + user certificates).
2. Auto generates `attestation.h` (file containing the signing authority's public keys + list of clients in the federation) using CMake.

### CODE REVIEW

File: demo/python/basic/secure-xgboost-demo.py [hunk 1]
```diff
@@ -51,6 +51,11 @@
 # Save model to a file
 booster.save_model(HOME_DIR + "demo/python/basic/modelfile.model")
 
+print("Loading model")
+# Load model from file
```
Comment (chester-leung) [inline demo/python/basic/secure-xgboost-demo.py:55]: Do we support load model now? If I remember correctly, adding an enclave symmetric key should enable this?

**Review by chester-leung — COMMENTED** [review]
Thanks! One clarification:

Are `config/user*.crt` and `config/user*.pem` sample certificates and keys? If so should we move them to the `demo/data` directory? They seem to be used in the demos, though it likely doesn't matter if a user generates a new certificate and private key and overrides the old ones, since all demos read in certificates and keys.

My thought is that moving them to the `demo/data` directory might make them easier for the user to find.

And a question:
The helper scripts don't generate/output a public key for each client right? I don't think we currently use the public key on the client side in any of the demos, but I wonder if it might be good to write out the public key anyway.

File: demo/python/basic/secure-xgboost-demo.py (hunk 1 above)
Comment (podcastinator) [reply demo/python/basic/secure-xgboost-demo.py:55]: Exactly.

Comment (chester-leung) [reply demo/python/basic/secure-xgboost-demo.py:55]: Cool! Does the load model code still need some work, or can it be uncommented?

**Comment by podcastinator** [comment]
> Are `config/user*.crt` and `config/user*.pem` sample certificates and keys? If so should we move them to the `demo/data` directory? They seem to be used in the demos, though it likely doesn't matter if a user generates a new certificate and private key and overrides the old ones, since all demos read in certificates and keys.
> 
> My thought is that moving them to the `demo/data` directory might make them easier for the user to find.

Yes, they're samples. I let them be in the config directory because that's where the helper scripts that generate the credentials are located as well. 
Not sure moving them elsewhere because: if you rebuild the repo and want to generate more / different user keys using the helpers, then you have to copy them to the new location as well. So things might be better consolidated this way.


> 
> And a question:
> The helper scripts don't generate/output a public key for each client right? I don't think we currently use the public key on the client side in any of the demos, but I wonder if it might be good to write out the public key anyway.

The certificate contains the public key, so I suppose there's no need to write it to a different file...

**Comment by chester-leung** [comment]
> > And a question:
> > The helper scripts don't generate/output a public key for each client right? I don't think we currently use the public key on the client side in any of the demos, but I wonder if it might be good to write out the public key anyway.
> 
> The certificate contains the public key, so I suppose there's no need to write it to a different file...

Right. My original thinking was that the user might want to see or use the public key itself without having to parse the certificate, but there's no need for that right now. We can add it later if needed.

File: demo/python/basic/secure-xgboost-demo.py (hunk 1 above)
Comment (podcastinator) [reply demo/python/basic/secure-xgboost-demo.py:55]: It works, uncommenting it.

**Review by chester-leung — APPROVED** [review]

**Comment by chester-leung** [comment]
Thanks!

**Pull request merged**